controllers/controllers.js

```diff
@@ -70,10 +73,12 @@ exports.getArticles = (req, res, next) => {
 }
 
 exports.postComment = (req, res, next) => {
-    const { article_id } = req.params
-    const { comment } = req.body
-    insertComment(article_id, comment).then((comment) => {
+    const  article_id  = req.params.article_id
+    const  comment  = req.body
+    const username = req.body.username
+    Promise.all([checkArticleIdExists(article_id), checkUserExists(username), insertComment(comment, article_id)]).then(([, , comment]) => {
```

Comment:
If you switch the insertComment to be the first in the array, you wont need the two commas - might look slightly cleaner

Reply:
Have tried this but it breaks the tests and could not find a solution.

Reply:
I feel like this isn't a good case to use a Promise.all what we want is to do the util function checks one at a time and then do the insert if no errors are thrown. More of a case for a chain of thens
Saying that though I'm happy for you to keep this code, always interesting to see the different approaches